### PR TITLE
fix(formselect): adds isRequired prop

### LINKED ISSUE #1397: Form - required attribute missing for some of the components
All of the form inputs that can have the `required` attribute should be enabled to have it with the `isRequired` prop.

This attribute is supported on `<select>`, `<textarea>`, and several types of `<input>` ([Note on mdn about what `<input>` types don't support `required`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-required))

### PR DESCRIPTION
**Pull request by jenny-s51**
Closes #1397

### CODE REVIEW

**Comment by tlabaj** [comment]
@jenny-s51 Why is this labeled breaking change?

**Comment by jenny-s51** [comment]
per @tlabaj , this is not considered a breaking change as it adds an optional (but not required) prop

File: packages/patternfly-4/react-core/src/components/FormSelect/FormSelect.tsx [hunk 1]
```diff
@@ -48,14 +51,15 @@ export class FormSelect extends React.Component<FormSelectProps> {
   };
 
   render() {
-    const { children, className, value, isValid, isDisabled, ...props } = this.props;
+    const { children, className, value, isValid, isDisabled, isRequired, ...props } = this.props;
     return (
       <select
         {...props}
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/FormSelect/FormSelect.tsx]: We should move `{...props}` to the end to not break consumers using the `required` property instead.

**Comment by jgiardino** [comment]
I can't say I understand what @redallen is suggesting 😉 so his suggestion might be really good and something worth considering, but when I review the updates in this PR, the form elements now work as I expect. I.e. I can add `isRequired` to `<FormSelect>` and see a `required` attribute on the `<select>` element.

As future work, it would be nice to have examples that clearly illustrate how to create required elements in a form (especially for the select element, because creating the options for a required select can be tricky), and also refine the strings so that they more accurately describe exactly what effect the `isRequired` prop will have. But I'm also curious what @redallen is suggesting and if that might affect what I would suggest in an issue.

**Comment by redallen** [comment]
I'm suggesting that if we have the following JSX:
```jsx
<FormGroup isRequired>
  <FormSelect />
</FormGroup>
```
that we automatically make `<FormSelect>` have `isRequired` passed to it since it's in a `FormGroup` that has `isRequired` set. I thought wrong, though!

**Comment by jgiardino** [comment]
> ...that we automatically make <FormSelect> have isRequired passed to it since it's in a FormGroup that has isRequired set. I thought wrong, though!

Ah, thanks for clarifying! I'm not opposed to this suggestion at all. I would suggest capturing this as a separate issue, though, assuming it's something others agree would be good to implement.

**Pull request merged**